File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxi = -float('inf')
        window_sum = 0
        n = len(nums)
        ## first window
        for i in range(k):
            window_sum += nums[i]
        maxi = window_sum/k

        # slide
        for i in range(k,n):
            window_sum -= nums[i-k]
            window_sum += nums[i]
            maxi = max(maxi, window_sum/k)
        return maxi

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        double maxi = INT_MIN;
        int window_sum = 0;

        // first window
        for(int i=0; i<k; i++){
            window_sum += nums[i];
        }
        maxi = (double)window_sum/k;

        // slide
        for (int i=k; i<n; i++){
            window_sum -= nums[i-k];
            window_sum += nums[i];
            maxi = max(maxi, (double)window_sum/k);
        }
        return maxi;
    }
};